src/query/src/promql/planner.rs

```diff
@@ -2219,6 +2219,17 @@ impl PromPlanner {
                 .context(DataFusionPlanningSnafu)?;
         }
 
+        ensure!(
+            left_context.field_columns.len() == 1,
+            MultiFieldsNotSupportedSnafu {
+                operator: "AND operator"
+            }
+        );
+        // Update the field column in context.
+        // The AND/UNLESS operator only keep the field column in left input.
+        let left_field_col = left_context.field_columns.first().unwrap();
+        self.ctx.field_columns = vec![left_field_col.clone()];
```

Comment:
Do we need to update the tag columns in `self.ctx`?

Reply:
The tag columns are not used in the subsequent steps, so I didn't update them.